src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -200,12 +223,12 @@ protected function doGenerate(array $variables, array $defaults, array $requirem
                         return '';
                     }
 
-                    $url = $token[1].$mergedParams[$varName].$url;
+                    $url = $token[1].self::encodeVariable($mergedParams[$varName]).$url;
```

Comment:
I didn't check: why isn't `$token[1]` encoded in some way?

Reply:
Good point. Had that in mind but did not care for now as this was not covered by any test...

`$token[1]` should most likely handled as decoded so we would need to encode here.

TODO: add test for encoding `$path` from `Route` to document intended bahviour.